src/studio/src/designer/backend/Services/Implementation/RepositorySI.cs

```diff
@@ -1163,6 +1168,70 @@ public async Task<RepositoryClient.Model.Repository> CreateService(string org, S
             return repository;
         }
 
+        /// <inheritdoc/>
+        public async Task<RepositoryClient.Model.Repository> CopyRepository(string org, string sourceRepository, string targetRepository, string developer)
+        {
+            var options = new RepositoryClient.Model.CreateRepoOption(targetRepository);
+
+            RepositoryClient.Model.Repository repository = await CreateRepository(org, options);
```

Comment:
Could consider renaming the CreateRepository method called here to CreateRemoteRepository to be more explicit and avoid confusion with all the other repositories we have.